codec-http2/src/main/java/io/netty/handler/codec/http2/WeightedFairQueueByteDistributor.java

```diff
@@ -550,16 +550,30 @@ void removeChild(State child) {
                 events.add(new ParentChangedEvent(child, child.parent));
                 child.setParent(null);
 
-                // Move up any grand children to be directly dependent on this node.
-                Iterator<IntObjectMap.PrimitiveEntry<State>> itr = child.children.entries().iterator();
-                while (itr.hasNext()) {
-                    takeChild(itr, itr.next().value(), false, events);
+                if (!child.children.isEmpty()) {
+                    // Move up any grand children to be directly dependent on this node.
+                    Iterator<IntObjectMap.PrimitiveEntry<State>> itr = child.children.entries().iterator();
+                    long totalWeight = child.getTotalWeight();
+                    do {
+                        // Redistribute the weight of child to its dependency proportionally.
+                        State dependency = itr.next().value();
+                        dependency.weight = (short) ((int) dependency.weight * child.weight / totalWeight);
```

Comment:
Ensure the result isn't less than 1 (e.g. `Math.max(1, ...)`)

https://datatracker.ietf.org/doc/html/rfc7540#section-5.3.2
>  All dependent streams are allocated an integer weight between 1 and
   256 (inclusive).

Reply:
sure, this is necessary